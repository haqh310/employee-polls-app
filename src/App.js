//<img src="../api/files/home/workspace/employee-polls-app/public/logo192.png" className="App-logo" alt="logo" />
import './App.css';
import {Routes, Route} from "react-router-dom" 
import {connect} from "react-redux";
import { useEffect, Fragment } from 'react';

import { handleInitialData } from "./actions/shared"
import Nav from "./components/Nav"
import Home from './components/Home';
import Login from './components/Login';
import NewQuestion from './components/NewQuestion';
import Leaderboard from './components/Leaderboard';
import Poll from "./components/Poll";
import Error404 from "./components/Error404";
import Private from './components/Private';

function App({dispatch, loggedIn}) {

  useEffect(() => {
    if(!loggedIn) {
      dispatch(handleInitialData())
    }
  })
    
    return (
      <Fragment>
        <div className="container">
            {loggedIn ? <Nav/> : ""}
            <Routes>
              <Route path="/login" element={<Login/>} />
              <Route path="/" exact element={<Private><Home/></Private>} />
              <Route path="/leaderboard" element={<Private><Leaderboard/></Private>} />
              <Route path="/add" element={<Private><NewQuestion/></Private>} />
              <Route path="/question/:id" element={<Private><Poll/></Private>} />
              <Route path="/404" element={<Error404/>} />
            </Routes>
        </div>
    </Fragment>
    );
}

const mapStateToProps = ({authedUser}) => ({
  loggedIn: !!authedUser
})

export default connect(mapStateToProps)(App);
