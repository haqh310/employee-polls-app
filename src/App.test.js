import { render} from '@testing-library/react';
import { store } from "./store";
import {BrowserRouter as Router} from "react-router-dom";
import { Provider } from 'react-redux';
import App from './App';

describe('App', () => {
  it('renders component correctly', () => {
      const component = render(
      <Provider store={store}>
          <Router>
            <App />
          </Router>
      </Provider>
      )
      expect(component).toMatchSnapshot();
  });
} )
