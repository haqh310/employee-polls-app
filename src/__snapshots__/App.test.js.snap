// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="d-flex justify-content-center my-5"
        >
          <div
            class="card col-4"
          >
            <div
              class="card-body"
            >
              <h4
                class="text-center"
              >
                Employee Polls
              </h4>
              <div
                class="text-center"
              >
                <img
                  alt="employee"
                  class="img-fluid"
                  src="employee.jpg"
                />
              </div>
              <form>
                <div
                  class="d-flex justify-content-center"
                >
                  <select
                    class="form-select my-3 w-75"
                    data-testid="select"
                    id="select"
                  >
                    <option
                      data-testid="select-option"
                      disabled=""
                      id="select-option"
                      value=""
                    >
                      Select employee
                    </option>
                  </select>
                </div>
                <div
                  class="d-flex justify-content-center"
                >
                  <button
                    class="btn btn-primary mx-5"
                    data-testid="btnSubmit"
                    id="btnSubmit"
                    type="submit"
                  >
                    Submit
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="d-flex justify-content-center my-5"
      >
        <div
          class="card col-4"
        >
          <div
            class="card-body"
          >
            <h4
              class="text-center"
            >
              Employee Polls
            </h4>
            <div
              class="text-center"
            >
              <img
                alt="employee"
                class="img-fluid"
                src="employee.jpg"
              />
            </div>
            <form>
              <div
                class="d-flex justify-content-center"
              >
                <select
                  class="form-select my-3 w-75"
                  data-testid="select"
                  id="select"
                >
                  <option
                    data-testid="select-option"
                    disabled=""
                    id="select-option"
                    value=""
                  >
                    Select employee
                  </option>
                </select>
              </div>
              <div
                class="d-flex justify-content-center"
              >
                <button
                  class="btn btn-primary mx-5"
                  data-testid="btnSubmit"
                  id="btnSubmit"
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
